### PR TITLE
fix: OOM in deepscaler1.5b with sequence length = 16/24k 

### LINKED ISSUE #846: [regression] Deepscaler requiring 4 nodes for 24k (used to require just 1)

### PR DESCRIPTION
**Pull request by soodoshll**
# What does this PR do ?

Address #846. Now the peak memory usage is ~60% for 16k, and ~80% for 24k.

Related #856 (there's some overlap and maybe need to merge manually)

# Issues

Fixes #846

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

### CODE REVIEW

**Comment by wangshangsam** [comment]
Bravos! @soodoshll could you elaborate how much GPU global memory footprint was saved?

File: nemo_rl/models/policy/dtensor_policy_worker.py [hunk 1]
```diff
@@ -1044,17 +1046,19 @@ def get_logprobs(
...
-                            logits.to(torch.float32),
+                            logits,
                             input_ids_dtensor,
                             seq_index_tensor,
                         )
 
                         assert token_logprobs.shape[1] == seq_len - 1
                     else:
                         if isinstance(logits, DTensor):
+                            logits = logits.to(torch.float32)
```
Comment (gshennvm) [inline nemo_rl/models/policy/dtensor_policy_worker.py:1059]: wondering why you moved this up? just a code cleanup?

Comment (soodoshll) [reply nemo_rl/models/policy/dtensor_policy_worker.py:1059]: I made this change to try to free the memory of the original logits asap. And this cast will be removed in #856.
